linkis-computation-governance/linkis-manager/linkis-application-manager/src/main/java/org/apache/linkis/manager/am/service/impl/DefaultEngineConnStatusCallbackService.java

```diff
@@ -0,0 +1,138 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.linkis.manager.am.service.impl;
+
+import org.apache.linkis.manager.am.conf.AMConfiguration;
+import org.apache.linkis.manager.am.service.EngineConnStatusCallbackService;
+import org.apache.linkis.manager.am.service.engine.EngineStopService;
+import org.apache.linkis.manager.common.constant.AMConstant;
+import org.apache.linkis.manager.common.entity.enumeration.NodeStatus;
+import org.apache.linkis.manager.common.entity.metrics.AMNodeMetrics;
+import org.apache.linkis.manager.common.protocol.engine.EngineConnStatusCallback;
+import org.apache.linkis.manager.common.protocol.engine.EngineConnStatusCallbackToAM;
+import org.apache.linkis.manager.common.protocol.engine.EngineStopRequest;
+import org.apache.linkis.manager.persistence.NodeMetricManagerPersistence;
+import org.apache.linkis.manager.service.common.metrics.MetricsConverter;
+import org.apache.linkis.rpc.Sender$;
+import org.apache.linkis.rpc.message.annotation.Receiver;
+import org.apache.linkis.server.BDPJettyServerHelper;
+
+import org.apache.commons.lang3.StringUtils;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.UnsupportedEncodingException;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+
+@Service
+public class DefaultEngineConnStatusCallbackService implements EngineConnStatusCallbackService {
+  private org.slf4j.Logger logger =
+      org.slf4j.LoggerFactory.getLogger(DefaultEngineConnStatusCallbackService.class);
+
+  @Autowired private NodeMetricManagerPersistence nodeMetricManagerPersistence;
+
+  @Autowired private MetricsConverter metricsConverter;
+
+  @Autowired private EngineStopService engineStopService;
+
+  private static final String[] canRetryLogs =
+      AMConfiguration.AM_CAN_RETRY_LOGS().getValue().split(";");
+
+  @Receiver
+  public void dealEngineConnStatusCallback(EngineConnStatusCallback protocol) {
+    logger.info(
+        "EngineConnStatusCallbackServiceImpl handle engineConnStatus callback serviceInstance: [{}] status: [{}]",
+        protocol.serviceInstance(),
+        protocol.status());
+    if (!NodeStatus.isAvailable(protocol.status())) {
+      EngineStopRequest engineStopRequest = new EngineStopRequest();
+      engineStopRequest.setServiceInstance(protocol.serviceInstance());
+      engineStopRequest.setUser("hadoop");
+      engineStopService.stopEngine(
```

Comment:
Need to add try catch to ensure the execution of dealEngineConnStatusCallbackToAM

Reply:
I have add try catch around stopEngine method to ensure the execution of dealEngineConnStatusCallbackToAM, thanks for your kind remind.

Reply:
> I have add try catch around stopEngine method to ensure the execution of dealEngineConnStatusCallbackToAM, thanks for your kind remind. Thanks for your time.

Thanks!